### PR TITLE
Rw refactor test suites

### PR DESCRIPTION
**Pull request by rwilliams659**
### Type of change made:
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
- [X] Refactoring

### Detailed Description
I went through each of the 5 test suites and refactored so that each was using a much smaller and more manageable test suite. I then went through each test and modified so each would pass with the modify test suite.

This also involved adding beforeEach hooks where they were not yet present, deleting unnecessary tests, and adding a few new ones. This should make the test suites easier to add to or modify moving forward if we need to as well. 

Additionally, a few issues with methods were identified during this process, and were fixed accordingly. All test suites were also renamed to reflect the new class names - This is why it appears there were so many changes. 

### Why is this change required? What problem does it solve?

This cleans up the test suites and makes them more readable. 

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?

There is still refactoring that needs to be done for the class methods themselves, which made figuring out whether they were working difficult in some cases. There were a few methods I couldn't quite figure out, and I left comments on those. With refactoring, we should be able to figure all that out. 

### Files modified:
- [ ] index.html
- [ ] style.scss
- [ ] scripts.js
- [X] Activity-repo.js
- [X] Hydration-repo.js
- [X] Sleep-repo.js
- [X] User-repo.js
- [X] User.js
- [ ] Other files:

##Relevant Ticket
Ticket #34

### CODE REVIEW

File: src/Activity-repo.js [hunk 1]
```diff
@@ -54,7 +55,7 @@ class ActivityRepo {
 
   accomplishStepGoalForDay(id, date, userRepo) {
     let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
-    if (userStepsByDate.numSteps === userRepo.dailyStepGoal) {
+    if (userStepsByDate.numSteps >= userRepo.dailyStepGoal) {
```
Comment (jkwest-93) [inline src/Activity-repo.js:58]: Nice catch!

Comment (H-Bo214) [reply src/Activity-repo.js:58]: Agreed!

File: src/Activity-repo.js [hunk 2]
```diff
@@ -73,21 +74,21 @@ class ActivityRepo {
   }
   //Finds a single user's activities, then returns the activity object with the greatest # of flightsOfStairs, which can be used to represent date of flightsOfStairs record
 
-  getAllUsersAverageDataForDay(date, userRepo, relevantData) {
+  getAllUsersAverageDataForDay(date, userRepo, dataKey) {
     let selectedDayData = userRepo.getAllUsersDayData(this.activityData, date);
-    return parseFloat((selectedDayData.reduce((acc, elem) => acc += elem[relevantData], 0) / selectedDayData.length).toFixed(1));
+    return parseFloat((selectedDayData.reduce((acc, elem) => acc += elem[dataKey], 0) / selectedDayData.length).toFixed(1));
```
Comment (H-Bo214) [inline src/Activity-repo.js:79]: DataKey is definitely easier to understand. Thanks

**Comment by jkwest-93** [comment]
This is great!! Thanks so much.

**Pull request merged**